sway-core/src/ir_generation/const_eval.rs

```diff
@@ -288,3 +310,180 @@ fn const_eval_typed_ast_node(
         TypedAstNodeContent::WhileLoop(_) | TypedAstNodeContent::SideEffect => None,
     }
 }
+
+pub fn serialize_to_storage_slots(
+    constant: &Constant,
+    context: &Context,
+    ix: &StateIndex,
+    ty: &Type,
+    indices: &[usize],
+) -> JsonStorageInitializers {
```

Comment:
This seems out of place in this file to me.  It's very storage/JSON specific, rather than 'constant evaluation' specific.

Reply:
Yeah I agree, I had a similar feeling. Generally that JSON stuff goes into the impl of the different types so it makes sense for this to go in Constant.rs but I didn't want to bloat sway-it with this stuff. So maybe a separate file in sway-core somewhere.